### PR TITLE
test: transpile modal ahead of tests

### LINKED ISSUE #32731: [Flaky CI]: customRepositories.spec.tsx: App Store Connect missing from page
### Flakiness Type

Assertion failure

### Name of Job

acceptance / frontend tests (0)

### Name of Test

tests/js/spec/views/settings/projectDebugFiles/customRepositories.spec.tsx

### Link to Test Run

https://github.com/getsentry/sentry/runs/5585976249?check_suite_focus=true#step:9:448

### Details

The change is a dev env change, thus, FE failures should be unexpected.

Failure snippet:
>      108 |     // Display modal content
>      109 |     // A single instance of App Store Connect is available on free plans
>      110 |     expect(await screen.findByText('App Store Connect credentials')).toBeInTheDocument();
>          |                         ^
>      111 |
>      112 |     // Close Modal
>      113 |     userEvent.click(screen.getByLabelText('Close Modal'));

### PR DESCRIPTION
**Pull request by Zylphrex**
The modals are being loaded during the test lazily. This change preloads the
modal in question in a before all so that the individual tests run fast.

Tries to fix https://github.com/getsentry/sentry/issues/32731

### CODE REVIEW

**Review by billyvg — APPROVED** [review]
Good find, let's add a TODO to figure out why it's slow.

File: tests/js/spec/views/settings/projectDebugFiles/customRepositories.spec.tsx [hunk 1]
```diff
@@ -88,6 +88,11 @@ describe('Custom Repositories', function () {
     type: CustomRepoType.APP_STORE_CONNECT,
   };
 
+  beforeAll(async function () {
+    // transpile the modal upfront so the test runs fast
+    await import('sentry/components/modals/debugFileCustomRepository');
+  });
```
Comment (evanpurkhiser) [inline tests/js/spec/views/settings/projectDebugFiles/customRepositories.spec.tsx:94]: What sort of speedup do we see?

Comment (Zylphrex) [reply tests/js/spec/views/settings/projectDebugFiles/customRepositories.spec.tsx:94]: potentially 4+ seconds in the test gets moved up to this `beforeAll`. The overall test suite will still take about the same time though.

Comment (Zylphrex) [reply tests/js/spec/views/settings/projectDebugFiles/customRepositories.spec.tsx:94]: here's a test run i ran to see how long this was taking https://github.com/getsentry/sentry/runs/5698279430?check_suite_focus=true#step:9:79

**Review by k-fish — APPROVED** [review]

**Pull request merged**